packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -192,6 +192,46 @@ export class SummarizerNode implements IRootSummarizerNode {
 		return { summary: result.summary, stats: result.stats };
 	}
 
+	public validateSummary() {
+		this.validateSummaryCore(false);
+	}
+
+	protected validateSummaryCore(parentSkipRecursion: boolean) {
+		assert(
+			this.wipSummaryLogger !== undefined,
+			0x1a3 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
+		);
+		assert(this.wipReferenceSequenceNumber !== undefined, 0x1a4 /* "Not tracking a summary" */);
+		let didNodeSummarize = this.wipLocalPaths !== undefined;
+
+		if (parentSkipRecursion) {
+			if (this._latestSummary !== undefined) {
+				didNodeSummarize = true;
+			} else {
+				return;
+			}
+		}
+
+		/**
+		 * The absence of wip local path indicates that summarize was not called for this node. This can happen if:
+		 * 1. A child node was created after summarize was already called on the parent. For example, a data store
+		 * is realized (loaded) after summarize was called on it creating summarizer nodes for its DDSes. In this case,
+		 * parentSkipRecursion will be true and the if block above would handle it.
+		 * 2. A new node was created but summarize was never called on it. This can mean that the summary that is
+		 * generated may not have the data from this node. We should not continue, log and throw an error. This
+		 * will help us identify these cases and take appropriate action.
+		 */
+		if (!didNodeSummarize) {
+			this.throwUnexpectedError({
+				eventName: "NodeNotSummarized",
+			});
```

Comment:
How big a loss is it to not have `proposalHandle` on this log?  That's some ID of the (to-be) uploaded summary right? Does it come from the server on successful upload or something?

Reply:
> We can add it to completeSummary though

Not sure what you mean. There's no logging or errors thrown in there that I can see

Reply:
Right. In validate step, we don't have the `proposalHandle` because the summary hasn't been uploaded yet. In complete step, we don't throw error anymore so we don't really care about `proposalHandle`.